frontend/device/sdl/device.lua

```diff
@@ -109,6 +109,7 @@ local Desktop = Device:new{
 
 local Emulator = Device:new{
     model = "Emulator",
+    hasDPad = no,
```

Comment:
Quoting @poire-z in https://github.com/koreader/koreader/pull/7183#issuecomment-765800280
> Possibly depending if both hasDPAD and hasKeys (or add hasKeyboard if hasKeys means something else) ?

There's also the issue that which behavior's best for the emulator may not be entirely the same as what's best for the AppImage/Desktop. I know I've often said a better emulator is a better desktop app (e.g., being able to drag files straight onto the window, being able to copy/paste, etc. just making it easier to quickly test things), but when we're talking about control basics like typing that's no longer necessarily as true.

Also note that we already have behavior like `local touchless = os.getenv("DISABLE_TOUCH") == "1"` (see line 138), easily triggered with `-t` in the `kodev` script. That doesn't actually disable touch, but perhaps it already covers what you're looking for? But of course I welcome improvements to the desktop app! But these keyboard type changes will require some careful thought to get right. :-)

Reply:
I get that, which is why it's an RFC. :-D

Basically, I just want the arrow keys to move the cursor and enter to add a new line (basically so I can just type into fields as normal), but not kill off any other arrow/enter usages. I'm unsure of the interplay between FocusManager, the VirtualKeyboard and InputText, the emulator and the app.

Reply:
> Basically, I just want the arrow keys to move the cursor and enter to add a new line 

And backspace to erase !
(Thank you for these keyboard tweaks, I had wished for that for a long time :)

Reply:
> And backspace to erase !

That should already work (and delete too)?

Reply:
Btw, I just noticed you can't focus on other input fields where there are multiple. (Select another one, types in the first.)

![Screenshot_20210123_194055](https://user-images.githubusercontent.com/202757/105610976-fa4f6080-5db2-11eb-9d1d-de602b86e194.png)

Reply:
> That should already work (and delete too)?

You're right.
It is not working for me because of my koreader/settings/event_map.lua containing:
```lua
return {
   [8] = "Back",
   [42] = "Back"
}
```
(If you have an idea on how I can continue having that, but still have it working in InputText, you'll avoid me looking for that :) if not, I'll look for it when I have time :)

Reply:
I think you'd expect back to… well, go back or close something. Overriding that in the keyboard seems like a breach of contract, but perhaps you could split it out into a separate keyboard back event, but then of course we'd need to be able to make one key to emit both at once.

Reply:
> Btw, I just noticed you can't focus on other input fields where there are multiple. (Select another one, types in the first.)
This appears to happen on master too?

Reply:
> I think you'd expect back to… well, go back or close something. Overriding that in the keyboard seems like a breach of contract,

"Back", yes, but overloading "Backspace" to mean "Go back" is also a bit surprising? Even Firefox is scrapping that as a default: https://www.zdnet.com/article/firefox-to-block-backspace-key-from-working-as-back-button/ (but yes, it's a _totally_ different UX in a web browser :-p)

Reply:
It's not overloaded, @poire-z just has a personal config.

I wasn't talking about this pr with the input focus thing. 😃